packages/financial-templates-lib/src/price-feed/CreatePriceFeed.js

```diff
@@ -174,6 +188,66 @@ async function createPriceFeed(logger, web3, networker, getTime, config) {
 
   // Internal helper methods:
 
+  // Returns an ExpressionPriceFeed.
+  async function _createExpressionPriceFeed(expressionConfig) {
+    // Build list of configs that could be used in the expression including default price feed configs and customFeeds
+    // that the user has provided inside the ExpressionPriceFeed config. Note: default configs are overriden by
+    // customFeeds with the same name. Tranform keys by escaping any special characters in the identifier names..
+    const allConfigs = Object.fromEntries(
+      Object.entries({ ...defaultConfigs, ...expressionConfig.customFeeds }).map(([key, value]) => {
+        return [escapeSpecialCharacters(key), value];
+      })
+    );
+
+    // This call chain:
+    // 1. Parses the expression into an expression tree of nodes.
+    // 2. Filters for "symbol" nodes, which would be price feed identifiers in this case.
+    // 3. Extract the name property for each of these symbol nodes
+    // 4. Puts it all in a set and converts back to an array to dedupe any repeated values.
+    const symbols = Array.from(
+      new Set(
+        math
+          .parse(expressionConfig.expression)
+          .filter(node => node.isSymbolNode)
+          .map(node => node.name)
+      )
+    );
+
+    // This is a complicated looking map that maps each symbol into an entry in an object with its value the price
+    // feed created from the mapped config in allConfigs.
+    const priceFeedMap = Object.fromEntries(
+      await Promise.all(
+        symbols.map(async symbol => {
+          const config = allConfigs[symbol];
+
+          // If there is no config for this symbol, insert null and send an error.
+          if (!config) {
+            logger.error({
+              at: "_createExpressionPriceFeed",
+              message: `No price feed config found for symbol: ${symbol} 🚨`,
+              expressionConfig
+            });
+            return [symbol, null];
+          }
+
+          // These configs will inherit the expression config values (except type), but prefer the individual config's
+          // value when present.
+          const combinedConfig = { ...expressionConfig, type: undefined, ...config };
+
+          // If this returns null, just return upstream since the error has already been logged and the null will be
+          // detected upstream.
+          const priceFeed = await createPriceFeed(logger, web3, networker, getTime, combinedConfig);
+          return [symbol, priceFeed];
+        })
+      )
+    );
+
+    // Return null if any of the price feeds in the map are null (meaning there was an error).
+    if (Object.values(priceFeedMap).some(priceFeed => priceFeed === null)) return null;
```

Comment:
if error was thrown earlier, i dont think this would be needed

Reply:
Agreed. However, the current pattern is to return null and log an error. I agree that this may be a bit nonsensical looking back, but I would prefer to update this pattern in a follow-up PR across this entire file rather than 1) being inconsistent or 2) creating a lot of unrelated churn by changing all of the creation functions.